ovh/resource_iam_resource_tags.go

```diff
@@ -0,0 +1,217 @@
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				ValidateDiagFunc: validateTags,
+			},
+		},
+	}
+}
+
+// validateTags validates tag keys and values according to the requirements
+func validateTags(v interface{}, path cty.Path) diag.Diagnostics {
+	var diags diag.Diagnostics
+	tags := v.(map[string]interface{})
+
+	keyRegex := regexp.MustCompile(`^[a-zA-Z0-9_.:/=+@-]{1,128}$`)
+	valueRegex := regexp.MustCompile(`^[a-zA-Z0-9_.:/=+@-]{0,256}$`)
+
+	for key, val := range tags {
+		if !keyRegex.MatchString(key) {
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Invalid tag key",
+				Detail:   fmt.Sprintf("Tag key '%s' must match pattern ^[a-zA-Z0-9_.:/=+@-]{1,128}$", key),
+			})
+		}
+
+		value := val.(string)
+		if !valueRegex.MatchString(value) {
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Invalid tag value",
+				Detail:   fmt.Sprintf("Tag value '%s' for key '%s' must match pattern ^[a-zA-Z0-9_.:/=+@-]{0,256}$", value, key),
+			})
+		}
+	}
+
+	return diags
+}
+
+func resourceIamResourceTagsCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*Config)
+	resourceURN := d.Get("resource_urn").(string)
+	tags := d.Get("tags").(map[string]interface{})
+
+	// Track successfully created tags for cleanup on error
+	createdTags := []string{}
+
+	// Create each tag
+	for key, val := range tags {
+		value := val.(string)
+		tagBody := map[string]string{
+			"key":   key,
+			"value": value,
+		}
+
+		endpoint := fmt.Sprintf("/v2/iam/resource/%s/tag", url.PathEscape(resourceURN))
+		err := config.OVHClient.PostWithContext(ctx, endpoint, tagBody, nil)
+		if err != nil {
+			// Best-effort cleanup: delete successfully created tags
+			for _, createdKey := range createdTags {
+				deleteEndpoint := fmt.Sprintf("/v2/iam/resource/%s/tag/%s", url.PathEscape(resourceURN), url.PathEscape(createdKey))
+				_ = config.OVHClient.DeleteWithContext(ctx, deleteEndpoint, nil)
+			}
+			return diag.FromErr(fmt.Errorf("failed to create tag '%s': %w", key, err))
+		}
+		createdTags = append(createdTags, key)
+	}
+
+	// Set the resource ID to the URN
+	d.SetId(resourceURN)
+
+	// Read back the resource to sync state
+	return resourceIamResourceTagsRead(ctx, d, meta)
+}
+
+func resourceIamResourceTagsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*Config)
+	resourceURN := d.Id()
+
+	// Fetch the resource details
+	var resourceDetails IamResourceDetails
+	endpoint := fmt.Sprintf("/v2/iam/resource/%s", url.PathEscape(resourceURN))
+	err := config.OVHClient.GetWithContext(ctx, endpoint, &resourceDetails)
+	if err != nil {
+		// If resource not found, remove from state
+		if errOvh, ok := err.(*ovh.APIError); ok && errOvh.Code == 404 {
+			d.SetId("")
```

Comment:
not sure what behavior you expect in this case ? 

If you do this, Terraform will try to create the `tags` resource again, but it won't work if the IAM resource doesn't exist. IMO we should just return an error in this case, but maybe I'm missing something ?

Reply:
That does make sense! I've removed this check to expose all errors to the user.